include/eve/arch/cpu/wide.hpp

```diff
@@ -451,43 +451,41 @@ namespace eve
         requires(!kumi::product_type<Type> && supports_bitwise_call<wide, V>)
 #endif
     {
-      return detail::self_bitor(w, o);
+      w = bit_or(w, o);
+      return w;
     }
 
     //! @brief Performs a bitwise or between all lanes of two wide instances.
     //! Do not participate to overload resolution if both wide doesn't has the same `sizeof`
     template<scalar_value U, typename M>
-    friend EVE_FORCEINLINE wide operator|(wide const& v, wide<U, M> const& w) noexcept
+    friend EVE_FORCEINLINE wide operator|(wide const& a, wide<U, M> const& b) noexcept
 #if !defined(EVE_DOXYGEN_INVOKED)
         requires(!kumi::product_type<Type> && supports_bitwise_call<wide, wide<U, M>>)
 #endif
     {
-      wide that = v;
-      return that |= w;
+      return bit_or(a, b);
     }
 
     //! @brief Performs a bitwise or between all lanes of a scalar and a eve::wide
     //! Do not participate to overload resolution if `sizeof(Type) != sizeof(S)`
     template<scalar_value S>
-    friend EVE_FORCEINLINE wide operator|(wide const& v, S w) noexcept
+    friend EVE_FORCEINLINE wide operator|(wide const& w, S s) noexcept
 #if !defined(EVE_DOXYGEN_INVOKED)
         requires(!kumi::product_type<Type> && supports_bitwise_call<wide, S>)
 #endif
     {
-      auto that = v;
-      return that |= bit_cast(w, as<Type> {});
+      return bit_or(w, bit_cast(s, as<Type>{}));
```

Comment:
These two free operators are inconsistent with the non free one. Let's make them consistent.
Ideally you also want to call `operator|=` as an implementation of `|`

Reply:
Inconsistent: there are 2 overloads of operator| while just one for operator|=

It's not a needless jump, it's a way to keep them consistent. You'd find it recommended in many places